### PR TITLE
Create function to calculate variability index

### LINKED ISSUE #60: Function to calculate the Variability Index
I have a function somewhere that calculates the variability index (VI).  I think this package would be a good home for it and I'd be happy to contribute it if there is a place for it.

The features module might be an appropriate place, but a daily float value for VI is returned rather than a boolean.

### PR DESCRIPTION
**Pull request by kanderso-nrel**
## Checklist

- [x] Closes #60
- [x] Added new API functions to `docs/api.rst`
- [x] Clearly documented all new API functions with [PEP257](https://www.python.org/dev/peps/pep-0257/) and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings
- [x] Adds description and name entries in the appropriate "what's new" file 
      in [`docs/whatsnew`](https://github.com/pvlib/pvanalytics/tree/master/docs/whatsnew) 
      for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` 
      or this Pull Request with `` :pull:`num` ``. Includes contributor name 
      and/or GitHub username (link with `` :ghuser:`user` ``).
- [x] Non-API functions clearly documented with docstrings or comments as necessary
- [x] Added tests to cover all new or modified code
- [x] Pull request is nearly complete and ready for detailed review

I'm tired of hunting down this function from some old script every time I want it... I'd rather just import it from pvanalytics instead :)

### CODE REVIEW

**Comment by wfvining** [comment]
Should probably add `@pytest.mark.xfail(strict=True)` to `test_clearsky.test_reno_identical()` and `test_clearsky.test_reno_begining_end()` until #105 is resolved.

**Comment by kanderso-nrel** [comment]
The coveralls step was throwing an error on 665d29a about `coveralls.exception.CoverallsException: Could not submit coverage: 422 Client Error: Unprocessable Entity for url: https://coveralls.io/api/v1/jobs`.  I can't pretend to know what the issue was or why my change fixed it, but here's the section of their docs that I copy/pasted from: https://coveralls-python.readthedocs.io/en/latest/usage/configuration.html#github-actions-support

Now that coveralls is running, I think the drop in coverage is due to a few lines in `test_reno_begining_end` not running, so not really related to this PR.

File: pvanalytics/metrics.py [hunk 1]
```diff
@@ -65,3 +65,43 @@ def performance_ratio_nrel(poa_global, temp_air, wind_speed, pac, pdc0,
...
+
+    Parameters
+    ----------
+    measured : Series
+        Time series of measured GHI. [W/m2]
+    clearsky : Series
+        Time series of the expected clearsky GHI. [W/m2]
+    freq : pandas datetime offset, optional
+        Aggregation period (e.g. 'D' for daily).  If not specified,
+        a single value of VI for the entire time series will be returned.
```
Comment (cwhanse) [inline pvanalytics/metrics.py:93]:
```suggestion
    freq : pandas datetime offset, optional
        Aggregation period (e.g. 'D' for daily).  If not specified,
        the variability index for the entire time series will be returned.
```

The default in `_calc_pathlength` is 'd', is 'd' as calendar day? I can't locate documentation for these strings.

**Review by cwhanse — APPROVED** [review]

File: pvanalytics/metrics.py (hunk 1 above)
Comment (kanderso-nrel) [reply pvanalytics/metrics.py:93]: Ah I meant to take that out.  I'll push another commit to remove the default value for that parameter. 

These magic string codes are documented in the table here: https://pandas.pydata.org/docs/user_guide/timeseries.html#dateoffset-objects

**Comment by wfvining** [comment]
Thanks for the coveralls fix, @kanderso-nrel. I'm not surprised that part of the pipeline went stale as support for Actions + Coveralls was still very new when I set it up in the first place. Agreed on the coverage drop, nothing to do with this.

**Comment by wfvining** [comment]
@kanderso-nrel sorry, this dropped off my radar. LGTM, I will go ahead and merge unless there is any objection.

**Review by wfvining — APPROVED** [review]

**Pull request merged**